src/vs/workbench/services/textfile/common/textFileEditorModelManager.ts

```diff
@@ -147,6 +152,10 @@ export class TextFileEditorModelManager implements ITextFileEditorModelManager {
 		return true;
 	}
 
+	public get onModelContentChanged(): Event<TextFileModelChangeEvent> {
```

Comment:
I see the need for an event when untitled or text files are changing so that the backup service can do its business, however I fear that this comes at the cost of spam: you will see many model content change events while the user is typing, so basically each key press will trigger this event. Performance should always be something to be concerned of, especially when it is typing performance we are talking about. 

Some ideas:
* buffer the model content change event and only start to emit the change after a certain delay
* do not always create an event object to send around

I think buffering would be a good idea anyway because backups should also not be created without some delay.